gcsfs/core.py

```diff
@@ -296,13 +328,38 @@ def _call(self, method, path, *args, **kwargs):
         meth = getattr(requests, method)
         if args:
             path = path.format(*[quote_plus(p) for p in args])
-        r = meth(self.base + path, headers=self.header, params=kwargs,
-                 json=json)
+        for retry in range(RETRY_COUNT):
+            try:
+                time.sleep(2**retry-1)
+                r = meth(self.base + path, headers=self.header, params=kwargs,
+                         json=json)
+                validate_response(r, path)
+                break
+            except RuntimeError as e:
+                if retry + 1 == RETRY_COUNT:
+                    raise e
+                html_error = _asHtmlError(e)
+                if html_error :
+                    if html_error['code'] in [500, 503, 504, '500', '503', '504']:
+                        # 5xx error, let's retry
+                        continue
+                raise e
+            except (requests.exceptions.ChunkedEncodingError,
```

Comment:
perhaps call this retriable_exceptions and move to utils module.

Reply:
> So runtime error is produced by requests?
If it's the one raised in validate_response (which I think it must be), then r already is defined and will always have the status_code attribute available - to avoid having to parse the contents.

I *think* you're trying to tell me to use the parsed data that is visible to validate_response instead of trying to parse that data again after validate_response unhelpfully serializes it to a string. If so then I agree and I think that the best way to do that is to raise an appropriate exception that contains the info we need.